config/environments/development.rb

```diff
@@ -1,8 +1,8 @@
 Rails.application.configure do
   # Settings specified here will take precedence over those in config/application.rb.
 
-  # In the development environment your application's code is reloaded on
-  # every request. This slows down response time but is perfect for development
+  # In the development environment your application's code is reloaded any time
+  # it changes. This slows down response time but is perfect for development
```

Comment:
Most of these changes came from me generating a fresh Rails 7 project and picking out the new defaults that look good. See the new template here https://github.com/rails/rails/blob/0add5dba834f2f1b84fcf1bd1b758545b325fb73/railties/lib/rails/generators/rails/app/templates/config/environments/development.rb.tt